ci_tutorial/tutorial.rst

```diff
@@ -0,0 +1,127 @@
+.. _ci_tutorial:
+
+Continuous Integration (CI) tutorial
+====================================
+
+.. note::
+
+  - This is an advanced topic, previous knowledge of Conan tool is necessary. Please :ref:`read and practice the user tutorial<tutorial>` first.
+  - This section is intended for devops and build engineers designing and implementing a CI pipeline involving Conan packages, if it is not the
+    case, you can skip this section.
+
+
+Continuous Integration has different meanings for different users and organizations. In this tutorial we will cover the scenarios when users
+are doing changes to the source code of their packages and want to automatically build new binaries for those packages and also compute if those new package changes integrate cleanly or break the organization main products.
+
+We will use in this tutorial this small project that uses several packages (static libraries by default) to build a couple of applications, a video game and a map viewer utility. The ``game`` and ``mapviewer`` are our final "**products**", what we distribute to our users:
+
+.. graphviz::
+    :align: center
+
+    digraph game {
+        node [fillcolor="lightskyblue", style=filled, shape=box]
+        rankdir="BT"
+        "game/1.0" -> "engine/1.0" -> "ai/1.0" -> "mathlib/1.0";
+        "engine/1.0" -> "graphics/1.0" -> "mathlib/1.0";
+        "mapviewer/1.0" -> "graphics/1.0";
+        "game/1.0" [fillcolor="lightgreen"];
+        "mapviewer/1.0" [fillcolor="lightgreen"];
+        {
+            rank = same;
+            edge[ style=invis];
+            "game/1.0" -> "mapviewer/1.0" ;
+            rankdir = LR;
+        }
+    }
+
+
+All of the packages in the dependency graph have a ``requires`` to its direct dependencies using version ranges, for example, ``game`` contains a ``requires("engine/[>=1.0 <2]")`` so new patch and minor versions of the dependencies will automatically be used without needing to modify the recipes.
+
+.. note::
+
+    **Important notes**
+
+    - This section is written as a hands-on tutorial. It is intended to be reproduced by copying the commands in your machine.
+    - The tutorial presents some of the tools, good practices and common approaches to the CI problem. But there are no silver bullets.
+      This tutorial is not the unique way that things should be done.
+      Different organizations might have different needs and priorities, different build services power and budget, different sizes, etc.
+      The principles and practices presented in the tutorial might need to be adapted.
```

Comment:
Maybe here in the same line of "your millage may vary", it's a good place to remind readers that they can contact us by opening an issue in gh if they have question on the approach for their use-case?

Reply:
Done